activation/handler.go

```diff
@@ -572,66 +580,78 @@ func (h *Handler) handleAtx(ctx context.Context, expHash types.Hash32, peer p2p.
 		select {
 		case err := <-ch:
 			h.log.WithContext(ctx).With().Debug("atx processed in other task", atx.ID(), log.Err(err))
-			return err
+			return nil, err
 		case <-ctx.Done():
-			return ctx.Err()
+			return nil, ctx.Err()
 		}
 	}
 
 	h.inProgress[atx.ID()] = []chan error{}
 	h.inProgressMu.Unlock()
 	h.log.WithContext(ctx).With().Info("handling incoming atx", atx.ID(), log.Int("size", len(msg)))
 
-	err := h.processAtx(ctx, expHash, peer, atx)
+	proof, err := h.processATX(ctx, expHash, peer, atx)
 	h.inProgressMu.Lock()
 	defer h.inProgressMu.Unlock()
 	for _, ch := range h.inProgress[atx.ID()] {
 		ch <- err
 		close(ch)
 	}
 	delete(h.inProgress, atx.ID())
-	return err
+	return proof, err
```

Comment:
proof, nil ?

Reply:
`h.processATX()` call above that may return an error, and that error needs to be sent over an "inProgress" channel, but it also needs to be returned so that `HandleSyncedAtx()` and `HandleGossipAtx()` don't swallow errors